vision-and-language/clip/README.md

```diff
@@ -0,0 +1,64 @@
+# NNabla-CLIP
+
+This repository is a nnabla implementation of CLIP released by openAI.
+
+## Approach
+
+![CLIP](CLIP.png)
+
+## Usage
+
+you can setup the environment via following command:
+```bash
+$ pip install -r requirements.txt
+```
+
+```python
+import nnabla as nn
+import nnabla.functions as F
+from PIL import Image
+
+import clip
+
+with nn.auto_forward():
+    clip.load('data/ViT-B-32.h5')
+
+    image = clip.preprocess(Image.open("CLIP.png"))
+    text = clip.tokenize(["a diagram", "a dog", "a cat"])
+
+    image_features = clip.encode_image(image)
+    text_features = clip.encode_text(text)
+            
+    logits_per_image, logits_per_text = clip.logits(image, text)
+    probs = F.softmax(logits_per_image, axis=-1)
+
+    print("Label probs:", probs.d)  # prints: [[0.9927937  0.00421068 0.00299572]]
+```
+
+## API
+
+The CLIP module `clip` provides the following methods:
+
+#### `clip.available_models()`
+
+Returns the names of the available CLIP models.
+
+#### `clip.load(path)`
+
+WIP
+
+#### `clip.tokenize(text: Union[str, List[str]], context_length=77)`
+
+Returns a NdArray containing tokenized sequences of given text input(s). This can be used as the input to the model
+
+#### `clip.encode_image(image: numpy.ndarray)`
```

Comment:
Currently, batch_size seems to be determined by how you pass input to encode_image.
You should either give an example here very clearly, or
set arg.batch_size for users and add an assert checking that the user input to encode_image has a compatible shape with batch_size.